src/elements/common/media-query/types.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+
+import { POINTER_TYPE_NONE, POINTER_TYPE_FINE, POINTER_TYPE_COARSE } from './constants';
+
+export type MediaPointerType = POINTER_TYPE_NONE | POINTER_TYPE_FINE | POINTER_TYPE_COARSE;
+export type MediaHoverType = 'none' | 'hover';
+
+export type MediaFeatures = {
+    anyHover?: MediaHoverType,
+    anyPointer?: MediaPointerType,
+    hover?: MediaHoverType,
+    maxHeight?: number,
+    maxWidth?: number,
+    minHeight?: number,
+    minWidth?: number,
+    pointer?: MediaPointerType,
+};
+
+export type MediaShape = {
+    anyHover: 'none' | 'hover',
+    anyPointer: 'none' | 'fine' | 'coarse',
+    hover: 'none' | 'hover',
+    pointer: 'none' | 'fine' | 'coarse',
+    size: boolean,
```

Comment:
should be an enum of strings 

do we also want to provide the exact screen dimensions here, or anything else?

Reply:
i've added view height and width